dbt/include/databricks/macros/materializations/incremental/strategies.sql

```diff
@@ -49,26 +49,34 @@
 {% endmacro %}
 
 {% macro get_insert_into_sql(source_relation, target_relation) %}
+    {%- set source_columns = adapter.get_columns_in_relation(source_relation) | map(attribute="quoted") | list -%}
+    {%- set dest_columns = adapter.get_columns_in_relation(target_relation) | map(attribute="quoted") | list -%}
+    {{ insert_into_sql_impl(target_relation, dest_columns, source_relation, source_columns) }}
+{% endmacro %}
 
+{% macro insert_into_sql_impl(target_relation, dest_columns, source_relation, source_columns) %}
     {%- set common_columns = [] -%}
-    {%- set source_columns = adapter.get_columns_in_relation(source_relation) | map(attribute="name") -%}
-    {%- for dest_col in adapter.get_columns_in_relation(target_relation) -%}
-      {% if dest_col.name in source_columns %}
-          {%- if common_columns.append(dest_col) -%}{%- endif -%}
+    {%- for dest_col in dest_columns -%}
+      {%- if dest_col in source_columns -%}
+        {%- do common_columns.append(dest_col) -%}
+      {%- else -%}
+        {%- do common_columns.append('NULL') -%}
```

Comment:
Hi, This might cause an explicit NULL insertion into the destination table and skip default values if set in schema ?

Reply:
Good call out.  I will add a test for this, and if necessary, spin a new release with the fix.

Reply:
The docs I came across said that the number of columns were assumed to match the target table, since they are not matched by name, but by column order.  I've switched to using DEFAULT in my new PR, which I think covers your concern.  And you're right, this is a feature we're looking into adding this year (full implementation of the column spec across all materializations).

Reply:
One more thing, with this feature request, we'll now expect/restrict users to always have aliases in their models.!

```sql
-- A valid code dbt-databricks supports till now, since there is no column check in source/destination:
create or replace temporary view test_view__dbt as
  select cast('002' as int), 'Test name';

insert into `test_db`.`test_table` -- (Schema: id, name)
  select * from test_view__dbt;
```

If this is known, below code might work for most of the cases:

```jinja
{% macro get_insert_into_sql(source_relation, target_relation) %}
    {%- set source_columns = adapter.get_columns_in_relation(source_relation) | map(attribute="quoted") -%}
    {%- set dest_columns = adapter.get_columns_in_relation(target_relation) | map(attribute="quoted") -%}    
    {%- set common_columns = set(source_columns).intersection(dest_columns) -%}
    
    {%- set common_columns_csv = common_columns | join(', ') -%}

    insert into table {{ target_relation }} ({{ common_columns_csv }})
      select {{common_columns_csv}} from {{ source_relation }}
{% endmacro %}
```

Benefits:
1. Number of columns & the order in source and target - will be same
2. No unexpected overrides - default feature will work
3. The size of source and destination columns won't matter for iteration performance
4. Will return only the unique ones - Though its never a possibility of having duplicate columns in source/target, but no harm too
5. Might be a faster - No conversion of generator or set to list, less space and control statements

Cons:
1. Sequence in common_columns output won't be in order - this won't impact the final query.

If looks good, please give it a try. Thanks

Reply:
hmm, interesting, let me look into this.